### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by hav-noms**

### CODE REVIEW

File: contracts/RewardEscrow.sol [hunk 1]
```diff
@@ -233,6 +233,7 @@ contract RewardEscrow is Owned, IRewardEscrow {
             }
             uint qty = getVestingQuantity(msg.sender, i);
             if (qty == 0) {
+                // This line can not be code covered as 0 quantities cant be created
```
Comment (justinjmoses) [inline contracts/RewardEscrow.sol:236]: why not refactor it to `if (qty > 0) { ... }`

File: contracts/SynthetixEscrow.sol [hunk 2]
```diff
@@ -223,6 +215,7 @@ contract SynthetixEscrow is Owned, LimitedSetup(8 weeks), IHasBalance {
             }
             uint qty = getVestingQuantity(msg.sender, i);
             if (qty == 0) {
+                // This line can not be code covered as 0 quantities cant be created
```
Comment (justinjmoses) [inline contracts/SynthetixEscrow.sol:218]: (ibid)

File: test/contracts/DelegateApprovals.js [hunk 3]
```diff
@@ -247,6 +247,84 @@ contract('DelegateApprovals', async accounts => {
...
+	describe('adding approvals for issueOnBehalf', async () => {
+		const authoriser = account1;
+		const delegate = account2;
+
+		it('should return false if no approval for account1', async () => {
+			const result = await delegateApprovals.canIssueFor(authoriser, delegate);
+			assert.isNotTrue(result);
+		});
+		it('should set approval for all approveIssueOnBehalf for account2', async () => {
+			await delegateApprovals.approveIssueOnBehalf(delegate, { from: authoriser });
```
Comment (justinjmoses) [inline test/contracts/DelegateApprovals.js:259]: can you also check the event was emitted?

File: test/contracts/DelegateApprovals.js [hunk 4]
```diff
@@ -247,6 +247,84 @@ contract('DelegateApprovals', async accounts => {
...
+	describe('adding approvals for burnOnBehalf', async () => {
+		const authoriser = account1;
+		const delegate = account2;
+
+		it('should return false if no approval for account1', async () => {
+			const result = await delegateApprovals.canBurnFor(authoriser, delegate);
+			assert.isNotTrue(result);
+		});
+		it('should set approval for all burnOnBehalf for account2', async () => {
+			await delegateApprovals.approveBurnOnBehalf(delegate, { from: authoriser });
```
Comment (justinjmoses) [inline test/contracts/DelegateApprovals.js:298]: (ibid)

File: test/contracts/DelegateApprovals.js [hunk 5]
```diff
@@ -247,6 +247,84 @@ contract('DelegateApprovals', async accounts => {
 		});
 	});
 
+	describe('adding approvals for issueOnBehalf', async () => {
```
Comment (justinjmoses) [inline test/contracts/DelegateApprovals.js:250]: You could look to refactor and dedupe all these tests like so:

```javascript
['Issue', 'Burn', 'Exchange', 'Claim'].forEach(type => {
   describe(`Adding approvals for ${type}`, () => {
      const fncs = { 
          check: `can${type}For`,
          approve: `approve${type}OnBehalf`
      };
      it ...
      it ...
      it ...

  });
});
```

Comment (justinjmoses) [reply test/contracts/DelegateApprovals.js:250]: also note that you don't need `async` in the `describe` block as there are no `await` calls inside the block directly.

File: test/contracts/EternalStorage.js [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+contract('EternalStorage', accounts => {
+	const EternalStorage = artifacts.require('EternalStorage');
+	const [deployerAccount, owner, associatedContract, account1] = accounts;
+	let eternalStorage;
+
+	const toBytes = key => w3utils.asciiToHex(key);
+
+	before(async () => {
+		({ EternalStorage: eternalStorage } = await setupAllContracts({
```
Comment (justinjmoses) [inline test/contracts/EternalStorage.js:21]: You could probably just use `setupContract` for this instead no?

File: test/contracts/EternalStorage.js [hunk 7]
```diff
@@ -0,0 +1,312 @@
+'use strict';
```
Comment (justinjmoses) [inline test/contracts/EternalStorage.js:1]: :100: !

File: contracts/RewardEscrow.sol (hunk 1 above)
Comment (hav-noms) [reply contracts/RewardEscrow.sol]: Done

File: contracts/SynthetixEscrow.sol (hunk 2 above)
Comment (hav-noms) [reply contracts/SynthetixEscrow.sol]: RFAL

File: test/contracts/EternalStorage.js (hunk 6 above)
Comment (hav-noms) [reply test/contracts/EternalStorage.js:21]: oh of course. Forgot we had that as an option

Comment (hav-noms) [reply test/contracts/EternalStorage.js]: RFAL

File: test/contracts/DelegateApprovals.js (hunk 5 above)
Comment (hav-noms) [reply test/contracts/DelegateApprovals.js:250]: Oh yeah this is nice

File: contracts/RewardEscrow.sol (hunk 1 above)
Comment (hav-noms) [reply contracts/RewardEscrow.sol]: RFAL

File: test/contracts/DelegateApprovals.js (hunk 3 above)
Comment (hav-noms) [reply test/contracts/DelegateApprovals.js:266]: RFAL

File: test/contracts/DelegateApprovals.js (hunk 5 above)
Comment (hav-noms) [reply test/contracts/DelegateApprovals.js]: Done

**Pull request merged**